internal/pkg/cli/svc_deploy.go

```diff
@@ -321,6 +321,27 @@ func (o *deploySvcOpts) configureContainerImage() error {
 	return nil
 }
 
+func (o *deploySvcOpts) getTopics(unmarshaledManifest interface{}) (map[string]string, error) {
+	type publish interface {
+		PublishCfg() *manifest.PublishConfig
+	}
+	mf, ok := unmarshaledManifest.(publish)
+	if !ok {
+		return nil, fmt.Errorf("%s does not have required method Publish()", o.name)
+	}
+
+	if mf.PublishCfg() == nil || mf.PublishCfg().Topics == nil {
+		return nil, nil
+	}
+	topics := make(map[string]string)
+	for _, topic := range mf.PublishCfg().Topics {
+		arn := "arn:aws:sns:" + o.targetEnvironment.Region + ":" + o.targetApp.AccountID + ":" + o.targetApp.Name + "-" + o.envName + "-" + aws.StringValue(topic.Name)
```

Comment:
I think this may break in China partition regions; the ARN in those regions is of the form `arn:aws-cn:sns:cn-north-1:...`. @iamhopaul123 any thoughts on the best way of localizing this?

Reply:
Also, can we refactor this construction to a call to `fmt.Sprintf(snsArnPattern, partition, region, accountID, app, env, svc, topicName)` and constantize the arn pattern to something like `arn:%s:sns:%s:%s:topic/%s-%s-%s-%s`?